lib/TPP/ConvInitSimplify.cpp

```diff
@@ -0,0 +1,113 @@
+//===- FastMathOpts.cpp.cpp --------------------------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Dialect/Tpp/TppUtils.h"
+#include "TPP/Passes.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Utils/Utils.h"
+#include "mlir/IR/AffineMap.h"
+#include "mlir/IR/Matchers.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+
+#define GEN_PASS_CLASSES
+#include "TPP/Passes.h.inc"
+
+namespace {
+
+static bool isBroadCastOp(linalg::GenericOp linalgOp) {
+  if (linalgOp->getNumOperands() != 2 || linalgOp->getNumResults() != 1)
+    return false;
+  if (!tpp::utils::hasOnlyYieldOp(linalgOp.getRegion()))
+    return false;
+  SmallVector<unsigned> perm;
+  return linalgOp.getMatchingIndexingMap(linalgOp.getDpsInputOperand(0))
+      .isPermutationOfMinorIdentityWithBroadcasting(perm);
+}
+
+static std::optional<linalg::GenericOp> getBroadCastProdcuer(OpOperand *rhs) {
+  linalg::GenericOp broadcastOp = rhs->get().getDefiningOp<linalg::GenericOp>();
+  if (!broadcastOp || !isBroadCastOp(broadcastOp))
+    return std::nullopt;
+  return broadcastOp;
+}
+
+static bool isConstantZero(linalg::FillOp fillOp) {
+  Value cst = fillOp.getInputs()[0];
+  return matchPattern(cst, m_AnyZeroFloat()) || matchPattern(cst, m_Zero());
+}
+
+static bool isFillWithZeros(linalg::Conv2DNhwcHwcfOp convOp) {
```

Comment:
Avoid specialized functions. Reuse existing `isZero`-type functions and get the value/op before calling them. 

This proliferation of similar functions lead to high maintenance cost and unexpected bugs when we expect the general function to be called and it isn't.

Reply:
Which function are you referring to? `isZeroTensor`?

Reply:
To begin with, your `isConstantZero` above already exists in `TppUtils` (`isValConstZero`).

But here, the `isZeroTensor` already checks to see if a tensor is filled with zeroes. It might not be *exactly* what you're looking for here, but duplicating logic in little static helpers all over the place won't help.

Please start commoning up those methods instead of proliferating them.

Reply:
`isZeroTensor` does not exist in `ConvInitSimplify.cpp`. There is `isFillWithZeros`. Is this the method you are referring to?

Reply:
Ok but you don't make me understand. In the specific case here, what is that you don't want?

Reply:
I want you to reuse the existing functions instead of creating new ones. To refactor existing functions to adapt to all previous and new uses, if they're not quite right.

In this particular case, find a way to adapt `isFilledWithZeros` to `isZeroTensor` (or whatever is more appropriate in utils).